### PR TITLE
etcdmain: unset ETCD_UNSUPPORTED_ARCH after arch check

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/9077.
ref. https://github.com/coreos/etcd/pull/9089#discussion_r159707772

### CODE REVIEW

File: pkg/flags/flag.go [hunk 1]
```diff
@@ -123,6 +123,10 @@ func verifyEnv(prefix string, usedEnvKey, alreadySet map[string]bool) {
 			plog.Warningf("recognized environment variable %s, but unused: shadowed by corresponding flag", kv[0])
 			continue
 		}
+		// unsupported architecture only configured via environment variable
+		if env == "ETCD_UNSUPPORTED_ARCH" {
```
Comment (xiang90) [inline pkg/flags/flag.go]: the flag pkg should not understand etcd specific thing. can we do something similar to usedEnvKey?

Comment (gyuho) [reply pkg/flags/flag.go]: Problem is that `usedEnvKey` is only populated when there is a matching *flag*. We don't have flag `unsupported-flag` in the flag set, so it warns.

Comment (xiang90) [reply pkg/flags/flag.go]: then how about cleaning `ETCD_UNSUPPORTED_ARCH` before we calling into this pkg in etcdmain or somewhere.

**Comment by gyuho** [comment]
@xiang90 PTAL.

**Comment by xiang90** [comment]
have you manually checked the issue is fixed?

**Comment by gyuho** [comment]
@xiang90 Yes. I manually commented out some logic in my amd64 machine, and confirm the expected behavior in this code path.

**Comment by xiang90** [comment]
lgtm

**Comment by hexfusion** [comment]
Thanks for effort here guys, I think this is a nicer way to handle the situation.

**Pull request merged**